extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -148,29 +151,39 @@ export class AmpAdApiHandler {
 
   /**
    * Updates the element's dimensions to accommodate the iframe's
-   *    requested dimensions.
+   * requested dimensions. Notifies the window that request the resize
+   * of success or failure.
    * @param {number|undefined} height
    * @param {number|undefined} width
+   * @param {!Window} source
+   * @param {string} origin
    * @private
    */
-  updateSize_(height, width) {
-    const targetOrigin =
-          this.iframe_.src ? parseUrl(this.iframe_.src).origin : '*';
-    this.baseInstance_.attemptChangeSize(height, width, () => {
-      postMessage(
-          this.iframe_,
-          'embed-size-changed',
-          {requestedHeight: height, requestedWidth: width},
-          targetOrigin,
-          this.is3p_);
-    }, () => {
-      postMessage(
-          this.iframe_,
-          'embed-size-denied',
-          {requestedHeight: height, requestedWidth: width},
-          targetOrigin,
-          this.is3p_);
-    });
+  updateSize_(height, width, source, origin) {
+    this.baseInstance_.attemptChangeSize(height, width,
+        () => this.sendEmbedSizeResponse_(
```

Comment:
`#attemptChangeSize` doesn't take a success and failure callback, it returns a Promise. You'll have to put these in a `#then`:

```js
attemptChangeSize(height, width).then(success, failure);